src/libraries/filcrypto/filproofs/algorithms.go

```diff
@@ -681,30 +700,34 @@ func (proof *SDRColumnProof) Verify(root []byte, challenge UInt) bool {
 	return proof.InclusionProof.Verify(root, challenge)
 }
 
-func (sdr *WinStackedDRG_I) CreateOfflineCircuitProof(challengeProofs []OfflineWindowChallengeProof, aux sector.ProofAuxTmp) sector.SealProof {
+func (sdr *WinStackedDRG_I) CreateOfflineCircuitProof(challengeProofs []OfflineWindowProof, aux sector.ProofAuxTmp) sector.SealProof {
 	// partitions := sdr.Partitions()
 	// publicInputs := GeneratePublicInputs()
 	panic("TODO")
 }
 
-func (sdr *WinStackedDRG_I) _generateOfflineChallenges(sealSeeds []sector.SealSeed, randomness sector.InteractiveSealRandomness, challengeCount WinStackedDRGChallenges, windowChallengeCount WinStackedDRGWindowChallenges) (challenges []UInt, windowChallenges []UInt) {
-	nodeSize := int(sdr.NodeSize())
-	nodes := sdr.Nodes()
+func (sdr *WinStackedDRG_I) _generateOfflineChallenges(sealSeed sector.SealSeed, randomness sector.InteractiveSealRandomness, wrapperChallengeCount WinStackedDRGChallenges, windowChallengeCount WinStackedDRGWindowChallenges) (windowChallenges []UInt, wrapperChallenges []UInt) {
+	wrapperChallenges = generateOfflineChallenges(int(sdr.Nodes()), sealSeed, randomness, int(wrapperChallengeCount))
+	windowChallenges = generateOfflineChallenges(int(sdr.WindowDRGCfg().Nodes()), sealSeed, randomness, int(windowChallengeCount))
+
+	return windowChallenges, wrapperChallenges
+}
 
-	challengeRangeSize := nodes - 1 // Never challenge the first node.
+func generateOfflineChallenges(challengeRange int, sealSeed sector.SealSeed, randomness sector.InteractiveSealRandomness, challengeCount int) []UInt {
+	var challenges []UInt
+	challengeRangeSize := challengeRange - 1 // Never challenge the first node.
 	challengeModulus := new(big.Int)
 	challengeModulus.SetUint64(uint64(challengeRangeSize))
 
-	count := int(challengeCount)
-	for i := 0; i < count; i++ {
-		var bytes []byte
-		for _, sealSeed := range sealSeeds {
-			bytes = append(bytes, sealSeed...)
-		}
-		bytes = append(bytes, randomness...)
-		bytes = append(bytes, littleEndianBytesFromInt(i, nodeSize)...)
+	// Maybe factor this into a separate function, since the logic is the same...
+
+	for i := 0; i < challengeCount; i++ {
+		var preimage []byte
+		preimage = append(preimage, sealSeed...)
+		preimage = append(preimage, randomness...)
+		preimage = append(preimage, littleEndianBytesFromInt(i, 4)...)
 
-		hash := HashBytes_SHA256Hash(bytes)
+		hash := HashBytes_SHA256Hash(preimage)
```

Comment:
DIFF: the current implementation does not truncate to Fr size

Reply:
I changed this, resolving the diff in favor of implementation. I'm now only truncating when generating labels. If the implementation does it elsewhere, please let me know, @dignifiedquire.